docs/dev/stream/operators/process_function.md

```diff
@@ -322,3 +322,26 @@ ctx.timerService.registerEventTimeTimer(coalescedTime)
 {% endhighlight %}
 </div>
 </div>
+
+### Fault Tolerance
+
+Timers registered within `ProcessFunction` are fault tolerant.
+
+Timers registered within `ProcessFunction` will be checkpointed by Flink. Upon restoring, timers that are checkpointed
+from the previous job will be restored on whatever new instance is responsible for that key.
+
+#### Processing Time Timers
+
+For processing timer timers, note that the firing time of a timer is an absolute value of when to fire.
+
+What this means is that if a checkpointed timer’s firing processing timestamp is t (which is basically the registering
+time + configured trigger time), then it will also fire at processing timestamp t on the new instance. Therefore, you
```

Comment:
What do you mean by `new instance`? Are you discussing the scenario when a task is recovered on a different machine? I don't think we need to mention this. It should be quite clear that clock synchronization is an issue in processing time. 

The info that a pt-timer fires on restore if the time passed while the job was down is important. Also mention that this is true for savepoint, which is even more critical because more time may pass between taking and restoring from a savepoint.